### ISSUE TITLE
Docs checklist

### ISSUE THREAD

**Issue by MaxHalford** [creme-ml/creme#235]
Here is a checklist of the classes and functions for which to vet the docstrings. We'll do this progressively for the `0.5.0` release.

**anomaly**

 - [ ] HalfSpaceTrees

**base**

 - [ ] AnomalyDetector
 - [ ] BinaryClassifier
 - [ ] Classifier
 - [ ] Clusterer
 - [ ] Ensemble
 - [ ] Estimator
 - [ ] MultiClassifier
 - [ ] MultiOutputClassifier
 - [ ] MultiOutputEstimator
 - [ ] MultiOutputRegressor
 - [ ] Proba
 - [ ] Regressor
 - [ ] Transformer
 - [ ] Wrapper

**cluster**

 - [ ] KMeans

**compat**

 - [ ] Creme2SKLClassifier
 - [ ] Creme2SKLClusterer
 - [ ] Creme2SKLRegressor
 - [ ] Creme2SKLTransformer
 - [ ] PyTorch2CremeRegressor
 - [ ] SKL2CremeClassifier
 - [ ] SKL2CremeRegressor
 - [ ] convert_creme_to_sklearn
 - [ ] convert_sklearn_to_creme

**compose**

 - [ ] Blacklister
 - [ ] FuncTransformer
 - [ ] Pipeline
 - [ ] Renamer
 - [ ] TransformerUnion
 - [ ] Whitelister

**datasets**

 - [ ] Airline
 - [ ] ChickWeights
 - [ ] CreditCard
 - [ ] Elec2
 - [ ] Higgs
 - [ ] KDD99HTTP
 - [ ] MaliciousURL
 - [ ] MovieLens100K
 - [ ] Phishing
 - [ ] Restaurants
 - [ ] SMSSpam
 - [ ] TREC07
 - [ ] ToulouseBikes
 - [ ] TrumpApproval

**dummy**

 - [ ] NoChangeClassifier
 - [ ] PriorClassifier
 - [ ] StatisticRegressor

**decomposition**

 - [ ] LDA

**ensemble**

 - [ ] AdaBoostClassifier
 - [ ] BaggingClassifier
 - [ ] BaggingRegressor
 - [ ] HedgeRegressor
 - [ ] StackingBinaryClassifier

**feature_extraction**

 - [ ] Agg
 - [ ] BagOfWords
 - [ ] Differ
 - [ ] TFIDF
 - [ ] TargetAgg

**feature_selection**

 - [ ] PoissonInclusion
 - [ ] SelectKBest
 - [ ] VarianceThreshold

**imblearn**

 - [ ] RandomOverSampler
 - [ ] RandomSampler
 - [ ] RandomUnderSampler

**impute**

 - [ ] PreviousImputer
 - [ ] StatImputer

**linear_model**

 - [ ] FFMClassifier
 - [ ] FFMRegressor
 - [ ] FMClassifier
 - [ ] FMRegressor
 - [ ] HOFMClassifier
 - [ ] HOFMRegressor
 - [ ] LinearRegression
 - [ ] LogisticRegression
 - [ ] PAClassifier
 - [ ] PARegressor
 - [ ] PoissonRegression
 - [ ] SoftmaxRegression

**meta**

 - [ ] BoxCoxRegressor
 - [ ] PredClipper
 - [ ] TransformedTargetRegressor

**metrics**

 - [ ] Accuracy
 - [ ] ClassificationReport
 - [ ] ConfusionMatrix
 - [ ] CrossEntropy
 - [ ] F1
 - [ ] FBeta
 - [ ] Jaccard
 - [ ] LogLoss
 - [ ] MAE
 - [ ] MCC
 - [ ] MSE
 - [ ] MacroF1
 - [ ] MacroFBeta
 - [ ] MacroPrecision
 - [ ] MacroRecall
 - [ ] Metric
 - [ ] MicroF1
 - [ ] MicroFBeta
 - [ ] MicroPrecision
 - [ ] MicroRecall
 - [ ] MultiFBeta
 - [ ] Precision
 - [ ] RMSE
 - [ ] RMSLE
 - [ ] ROCAUC
 - [ ] Recall
 - [ ] RegressionMultiOutput
 - [ ] Rolling
 - [ ] SMAPE
 - [ ] TimeRolling
 - [ ] WeightedF1
 - [ ] WeightedFBeta
 - [ ] WeightedPrecision
 - [ ] WeightedRecall

**model_selection**

 - [ ] expand_param_grid
 - [ ] progressive_val_score
 - [ ] successive_halving

**multiclass**

 - [ ] OneVsRestClassifier

**multioutput**

 - [ ] ClassifierChain
 - [ ] RegressorChain

**naive_bayes**

 - [ ] BernoulliNB
 - [ ] ComplementNB
 - [ ] GaussianNB
 - [ ] MultinomialNB

**neighbors**

 - [ ] KNeighborsClassifier
 - [ ] KNeighborsRegressor

**optim**

 - [ ] AMSGrad
 - [ ] AdaBound
 - [ ] AdaDelta
 - [ ] AdaGrad
 - [ ] AdaMax
 - [ ] Adam
 - [ ] FTRLProximal
 - [ ] MiniBatcher
 - [ ] Momentum
 - [ ] Nadam
 - [ ] NesterovMomentum
 - [ ] Optimizer
 - [ ] RMSProp
 - [ ] SGD

**preprocessing**

 - [ ] Binarizer
 - [ ] FeatureHasher
 - [ ] MaxAbsScaler
 - [ ] MinMaxScaler
 - [ ] Normalizer
 - [ ] OneHotEncoder
 - [ ] PolynomialExtender
 - [ ] RBFSampler
 - [ ] RobustScaler
 - [ ] StandardScaler

**proba**

 - [ ] Gaussian
 - [ ] Multinomial

**reco**

 - [ ] Baseline
 - [ ] BiasedMF
 - [ ] FunkMF
 - [ ] RandomNormal

**stats**

 - [ ] AbsMax
 - [ ] AutoCorrelation
 - [ ] BayesianMean
 - [ ] Bivariate
 - [ ] Count
 - [ ] Covariance
 - [ ] EWMean
 - [ ] EWVar
 - [ ] Entropy
 - [ ] IQR
 - [ ] Kurtosis
 - [ ] Max
 - [ ] Mean
 - [ ] Min
 - [ ] Mode
 - [ ] NUnique
 - [ ] PeakToPeak
 - [ ] PearsonCorrelation
 - [ ] Quantile
 - [ ] RollingAbsMax
 - [ ] RollingIQR
 - [ ] RollingMax
 - [ ] RollingMean
 - [ ] RollingMin
 - [ ] RollingMode
 - [ ] RollingPeakToPeak
 - [ ] RollingQuantile
 - [ ] RollingSEM
 - [ ] RollingSum
 - [ ] RollingVar
 - [ ] SEM
 - [ ] Skew
 - [ ] Sum
 - [ ] Univariate
 - [ ] Var

**stream**

 - [ ] DictReader
 - [ ] iter_array
 - [ ] iter_csv
 - [ ] iter_pandas
 - [ ] iter_sklearn_dataset
 - [ ] iter_vaex
 - [ ] shuffle

**time_series**

 - [ ] Detrender
 - [ ] GroupDetrender
 - [ ] SNARIMAX

**tree**

 - [ ] DecisionTreeClassifier
 - [ ] RandomForestClassifier

**utils**

 - [ ] Histogram
 - [ ] SDFT
 - [ ] Skyline
 - [ ] SortedWindow
 - [ ] Window

Here is the code I used for generating this list:

```python
import inspect
import importlib

def is_estimator(obj):
    return inspect.isclass(obj) or inspect.isfunction(obj)

for submodule in importlib.import_module('creme').__all__:

    print(f'**{submodule}**')
    print()
    
    for name, obj in inspect.getmembers(importlib.import_module(f'creme.{submodule}')):
        print(f' - [ ] {name}')
    print()
```
